pilot/pkg/model/config_test.go

```diff
@@ -282,6 +288,63 @@ func TestMostSpecificHostMatch(t *testing.T) {
 	}
 }
 
+func BenchmarkMostSpecificHostMatchExact(b *testing.B) {
+	length := 5000
+	m := make(map[host.Name]struct{})
+	var l []host.Name
+
+	for i := 0; i <= length; i++ {
+		h := host.Name(strconv.Itoa(i) + ".foo.bar.com")
+		l = append(l, h)
+		m[h] = struct{}{}
+	}
+
+	needle := host.Name(strconv.Itoa(length) + ".foo.bar.com")
+
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		_, _ = model.MostSpecificHostMatch(needle, m, l)
```

Comment:
same comment in other PR, shouldn't we construct the map in the loop?

Reply:
I think both slice and map were created (and slice has a sort operation) before a lot of calls to the MostSpecificMatch function. Writing once and reading many times. So is there no need to construct map (or slice) inside the loop here?

Reply:
In the istio code, slice and map are both created here: `istio/pilot/pkg/model/destination_rule.go:82`
```go
func (ps *PushContext) mergeDestinationRule(p *processedDestRules, destRuleConfig config.Config, exportToMap map[visibility.Instance]bool) {
        // ...
	// DestinationRule does not exist for the resolved host so add it
	p.hosts = append(p.hosts, resolvedHost)
	p.hostsMap[resolvedHost] = struct{}{}
	p.destRule[resolvedHost] = &destRuleConfig
	p.exportTo[resolvedHost] = exportToMap
}

```